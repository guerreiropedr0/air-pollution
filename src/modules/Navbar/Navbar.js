import { useNavigate } from 'react-router-dom';
import { AiOutlineLeft, AiOutlineClose } from 'react-icons/ai';
import { IoSettingsSharp } from 'react-icons/io5';
import { useState } from 'react';
import styles from './Navbar.module.css';

const Navbar = () => {
  const navigate = useNavigate();

  const [modalVisible, setModalVisible] = useState(false);
  const [theme, setTheme] = useState({
    color: ['Pink', 'Blue'],
    isPink: true,
  });

  const handleTheme = () => {
    if (theme.isPink) {
      document.documentElement.style.setProperty('--primary-color', '#4369b2');
      document.documentElement.style.setProperty(
        '--secondary-color',
        '#3f62a4',
      );
      document.documentElement.style.setProperty('--tertiary-color', '#3b5a99');
      document.documentElement.style.setProperty(
        '--quaternary-color',
        '#456bb2',
      );
    } else {
      document.documentElement.style.setProperty('--primary-color', '#eb4c89');
      document.documentElement.style.setProperty(
        '--secondary-color',
        '#de4682',
      );
      document.documentElement.style.setProperty('--tertiary-color', '#d4447b');
      document.documentElement.style.setProperty(
        '--quaternary-color',
        '#f44f8f',
      );
    }
    setTheme({ ...theme, isPink: !theme.isPink });
  };

  return (
    <nav className={styles['nav-bar']}>
      <ul className={styles['nav-links']}>
        <li>
          {window.location.href !== 'http://localhost:3000/' && window.location.href !== 'guerreiropedr0-air-pollution.netlify.app' && (
            <AiOutlineLeft
              className={styles['nav-icon']}
              onClick={() => navigate(-1)}
            />
          )}
        </li>
        <li className={styles['nav-item']}>2022 stats</li>
        <li>
          <IoSettingsSharp
            data-testid="icon"
            className={styles['nav-icon']}
            onClick={() => setModalVisible(!modalVisible)}
          />
        </li>
      </ul>
      {modalVisible && (
        <div className={styles.modal}>
          <div className={styles['modal-content']}>
            <button
              className={styles.theme}
              onClick={handleTheme}
              type="button"
            >
              Change Theme To
              {' '}
              {theme.isPink ? theme.color[1] : theme.color[0]}
            </button>
            <AiOutlineClose
              className={styles.close}
              onClick={() => setModalVisible(!modalVisible)}
            />
          </div>
        </div>
      )}
    </nav>
  );
};

export default Navbar;
