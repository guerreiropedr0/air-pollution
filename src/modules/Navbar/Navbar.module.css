.nav-bar {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1;
  background-color: var(--primary-color);
}

.nav-links {
  display: flex;
  justify-content: space-between;
  padding: 0.5rem;
}

.nav-item {
  opacity: 0.8;
}

.nav-icon:hover,
.nav-icon:active {
  transform: scale(1.2);
}

.modal {
  position: fixed;
  z-index: 1;
  padding-top: 100px;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: #000;
  background-color: #0006;
}

.modal-content {
  background-color: var(--secondary-color);
  margin: auto;
  padding: 1.25rem;
  width: 80%;
  display: grid;
  grid-template-columns: repeat(2, auto);
}

.close {
  background-color: transparent;
  justify-self: end;
  font-size: 28px;
}

.theme {
  background-color: var(--tertiary-color);
  border: 1px solid var(--white-color);
}
