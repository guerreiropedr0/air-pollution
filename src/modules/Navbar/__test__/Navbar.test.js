import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Navbar';

const MockedNavbar = () => (
  <BrowserRouter>
    <Navbar />
  </BrowserRouter>
);

describe('Navbar component', () => {
  it('renders succesfully', () => {
    render(<MockedNavbar />);
    const listedItem = screen.getByText('2022 stats');
    expect(listedItem).toMatchSnapshot();
  });

  it('theme icon changes theme', () => {
    render(<MockedNavbar />);
    const listedItem = screen.getByTestId('icon');
    fireEvent.click(listedItem);
    const button = screen.getByRole('button', 'Change Theme To Blue');
    fireEvent.click(button);
    expect(button.innerHTML).toBe('Change Theme To Pink');
  });
});
