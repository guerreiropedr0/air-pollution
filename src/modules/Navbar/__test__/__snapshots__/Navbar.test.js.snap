// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders succesfully 1`] = `
<li
  class="nav-item"
>
  2022 stats
</li>
`;
